lib/compile/csdl2openapi.js

```diff
@@ -2271,6 +2271,16 @@ see [Expand](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-prot
             s['x-sap-odm-oid-reference-entity-name'] = element['@ODM.oidReference'].entityName
         }
 
+        //for primitive types creating a wrapper object
+        if (element.$Type?.startsWith('Edm.')) {
```

Comment:
Should we not look at the schema return type instead of this `Edm.` check?

Reply:
yeah, made the changes.
Checked with Ralf's repo and have added similarly.